solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -874,14 +875,12 @@ public void removeLiveNodesListener(LiveNodesListener listener) {
   /**
    * Returns the lowest Solr version among all live nodes in the cluster.
    *
-   * @return the lowest Solr version as a String, or null if no version is found
-   * @throws KeeperException if a ZooKeeper error occurs
-   * @throws InterruptedException if the thread is interrupted
+   * @return the lowest Solr version as a String, or null if no version is found, throw Semver with
+   *     invalid version
```

Comment:
```suggestion
   * @return the lowest Solr version as a String, possibly null if <em>any</em> node doesn't have the version.
```
-- it's important to work this way because the caller needs to know if the whole cluster is at least a certain version.
What's unclear is what to do if there are no live nodes.  Null wouldn't make sense (as it would indicate an old Solr is running); LATEST (current software) would make sense.  Okay but now imagine that there are some live nodes of Solr 10.2 while the current code is 10.0 but has no live node.  It would probably make sense to consider the current code as the lowest version, even if it has no live node registered.  It's an edge case.

In essence, we're looking for a "smallestSolrVersionSeen", to include ourselves, looking at live nodes as well.  Null means older than 9.9 which is when this will be released.